File: packages/flora-bloom/src/types/dtos/file.ts
type FileUpload = {
  file: File;
};

type FileOnResponse = {
  id: number;
  createdAt: Date;
  updatedAt: Date;
  mimeType: string;
  key: string;
  size: number;
  eTag: string;
  originalName: string;
};

export type { FileUpload, FileOnResponse };

File: packages/flora-bridge/dto/file.go
package dto

import (
	"mime/multipart"
	"time"
)

type FileUpload struct {
	File *multipart.FileHeader `form:"file" binding:"required"`
}

type FileOnResponse struct {
	ID           uint      `json:"id" binding:"required"`
	CreatedAt    time.Time `json:"createdAt" binding:"required"`
	UpdatedAt    time.Time `json:"updatedAt" binding:"required"`
	MIMEType     string    `json:"mimeType" binding:"required"`
	Key          string    `json:"key" binding:"required"`
	Size         int64     `json:"size" binding:"required"`
	ETag         string    `json:"eTag" binding:"required"`
	OriginalName string    `json:"originalName" binding:"required"`
}
